File: Java/adivinarNumero.java
import java.util.Scanner;

public class adivinarNumero {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int val1;
        int numero = (int)(Math.random() * 100 + 1); 
        int cont = 0;

        System.out.println("JUEGO DE ADIVINAR LOS NÚMEROS");
        do {
            System.out.print("Ingresa un número: ");
            while (!sc.hasNextInt()) {
                System.out.println("¡Por favor, ingresa un número válido!");
                sc.next(); 
                System.out.print("Ingresa un número: ");
            }
            val1 = sc.nextInt(); 
            cont++;

            if (val1 < numero) {
                System.out.println("El número es mayor");
            } else if (val1 > numero) {
                System.out.println("El número es menor");
            }
        } while (val1 != numero);

        System.out.println("¡LO LOGRASTE!");
        System.out.println("Número de intentos: " + cont);

        sc.close();
    }
}

File: Python/adivinarNumero.py
import random
num = random.randint(1,100)
print(num)
a = 0
intentos = 0
while a != num:
    try: 
        intentos += 1 
        a = int(input("Ingresa un número entre 1 y 100: "))
        if a == num:
            print("FELICIDADES LO LOGRASTE!")
            print(f"Número de intentos: {intentos}")
        elif a>num:
            print("El número el menor")
        else:
            print("El número es mayor!")
    except ValueError: 
        print("Ingresa un valor valido!")

